include/deal.II/base/symmetric_tensor.h

```diff
@@ -2320,6 +2325,1003 @@ Number second_invariant (const SymmetricTensor<2,3,Number> &t)
+
+      // If vectors are nearly linearly dependent, or if there might have
+      // been large cancellations in the calculation of A[i][i] - w[0], fall
+      // back to QL algorithm
+      if (eig_vals_vecs[0].second * eig_vals_vecs[1].second > error)
+        {
+          return ql_implicit_shifts(A);
+        }
+
+      return eig_vals_vecs;
+    }
+
+
+
+    /**
+     * Compute the eigenvalues and eigenvectors of a real-valued rank-2
+     * symmetric tensor using the characteristic equation to compute eigenvalues
+     * and an analytical approach based on the cross-product for the eigenvectors.
+     * If the computations are deemed too inaccurate then the method falls
+     * back to ql_implicit_shifts.
+     * The specialized algorithm implemented here is given in
+     * Kopp, J.
+     * Efficient numerical diagonalization of hermitian 3x3 matrices
+     * International Journal of Modern Physics C, 2008, 19, 523-548
+     * doi: 10.1142/S0129183108012303
+     * arXiv.org preprint: physics/0610206
+     *
+     * @param[in] A The tensor of which the eigenvectors and eigenvalues are
+     * to be computed.
+     *
+     * @return An array containing the eigenvectors and the associated eigenvalues
+     */
+    template <typename Number>
+    std_cxx11::array<std::pair<Number, Tensor<1,3,Number> >,3>
+    hybrid (const dealii::SymmetricTensor<2,3,Number> &A)
+    {
+      static_assert(numbers::NumberTraits<Number>::is_complex == false,
+                    "This implementation of the 3d Hybrid algorithm does "
+                    "not support complex numbers");
+
+      const unsigned int dim = 3;
+      Number norm;          // Squared norm or inverse norm of current eigenvector
+      Number t, u;          // Intermediate storage
+
+      // Calculate eigenvalues
+      const std_cxx11::array<Number,dim> w = eigenvalues(A);
+
+      t = std::abs(w[0]);
+      for (unsigned int i=1; i<dim; ++i)
+        {
+          u = std::abs(w[i]);
+          if (u > t)
+            t = u;
+        }
+
+      if (t < 1.0)
+        u = t;
+      else
+        u = t*t;
+
+      // Estimated maximum roundoff error
+      const double error = 256.0 * std::numeric_limits<double>::epsilon() * u*u;
+
+      // Initialize the transformation matrix as the
+      // identity tensor
+      dealii::Tensor<2,dim,Number> Q;
+      Q[0][1] = A[0][1]*A[1][2] - A[0][2]*A[1][1];
+      Q[1][1] = A[0][2]*A[0][1] - A[1][2]*A[0][0];
+      Q[2][1] = A[0][1]*A[0][1];
+
+      // Calculate first eigenvector by the formula
+      //   v[0] = (A - w[0]).e1 x (A - w[0]).e2
+      Q[0][0] = Q[0][1] + A[0][2]*w[0];
+      Q[1][0] = Q[1][1] + A[1][2]*w[0];
+      Q[2][0] = (A[0][0] - w[0]) * (A[1][1] - w[0]) - Q[2][1];
+      norm    = Q[0][0]*Q[0][0] + Q[1][0]*Q[1][0] + Q[2][0]*Q[2][0];
+
+      // If vectors are nearly linearly dependent, or if there might have
+      // been large cancellations in the calculation of A[i][i] - w[0], fall
```

Comment:
Well, to be honest I don't know so if you say that it is the AE spelling then I trust that it is. I was just surprised that google preferred it spelt with one "l", and my word processor didn't care how it was spelt. I'll make the change :-)